train7,8以降  
潮位ラベルで2種類に分けて学習モデルを作成

# 4. run_testを使ったモデルの検証
予測モデルが作成できたら本番環境を模したrun_testを使ってモデルの検証を行います。  
run_testの操作はwindowsであればコマンドプロンプト、macであればターミナルで行います。  
!を使うことでnotebookのセル上での実行も可能ですので、本チュートリアルではこの方法で検証を行います。  

「1. 準備」より、ご自身のディレクトリ構造を再確認いただき、下記を参考に引数を設定し実行ください。  

readme.mdより  

引数"--exec-path"には実装した予測プログラム("predictor.py")が存在するパス名を指定する.  
引数"--data-dir"には配布された学習用データ"train"のパス名を指定する.  
引数"--start-date"には検証用データを作成するときの最初の日にちを0以上の整数値を指定する.  
引数"--end-date"には検証用データを作成するときの最後の日にちを0以上の整数値を指定する.  
"--start-date"で指定した値よりも大きい値を設定すること.  

In [1]:
!pip install lightgbm -q

### 参考(現状の最高精度のモデル)

```
# 学習4年分
# model_20221218-1.pkl
# 提出フォルダ : 20221218-1

start date: 1642, end date:2190
Loaded the model.   
100%|█████████████████████████████████████████| 549/549 [04:29<00:00,  2.04it/s]

RMSE: 0.12515357397949056
-----
# 学習5年分
# model_20221218-2.pkl
# 提出フォルダ : 20221218-2

start date: 1826, end date:2190
Loaded the model.   
100%|█████████████████████████████████████████| 365/365 [02:52<00:00,  2.12it/s]
RMSE: 0.09500319006972904
```

### train6-2 + 雨量追加

提出イメージ
```
1. 潮位ラベルのCSVも保存する
2. 水位(159), 水位(7)はそのまま1日ずらせば良い

・水位（159)
当日データ(3時間おき、mean,min,max,std)
1日前(3時間おき、mean,min,max,std)
2日前(3時間おき、mean,min,max,std)

・水位(7)
当日データ
1日前(3時間おき,mean,min,max,std)
2日前(3時間おき,mean,min,max,std)
3日前(3時間おき,mean,min,max,std)

雨量
当日(12時から3時間おきのmean,max、1日の平均)
1日前(1日の平均)
2日前(1日の平均)
```

In [6]:
# 学習4年分
# model_20221231-4-7.pkl, model_20221231-4-159.pkl
!python ../inference-test/run.py  --exec-path ../inference-test/src --data-dir ../data --start-date 1461 --end-date 2190


start date: 1461, end date:2190

Processing waterlevel data
done
Processing rainfall data
done
Processing tidelevel data
done
Making output data
done

Moved to /notebooks/inference-test/src

Loaded the model.   
100%|█████████████████████████████████████████| 730/730 [02:50<00:00,  4.29it/s]

RMSE: 0.11479712375601998


In [5]:
# 学習4年分
# model_20221231-4-7.pkl, model_20221231-4-159.pkl
!python ../inference-test/run.py  --exec-path ../inference-test/src --data-dir ../data --start-date 1642 --end-date 2190


start date: 1642, end date:2190

Processing waterlevel data
done
Processing rainfall data
done
Processing tidelevel data
done
Making output data
done

Moved to /notebooks/inference-test/src

Loaded the model.   
100%|█████████████████████████████████████████| 549/549 [02:07<00:00,  4.30it/s]

RMSE: 0.1185998215875185


In [4]:
# 学習4年分
# model_20221231-4-7.pkl, model_20221231-4-159.pkl
!python ../inference-test/run.py  --exec-path ../inference-test/src --data-dir ../data --start-date 1826 --end-date 2190


start date: 1826, end date:2190

Processing waterlevel data
done
Processing rainfall data
done
Processing tidelevel data
done
Making output data
done

Moved to /notebooks/inference-test/src

Loaded the model.   
100%|█████████████████████████████████████████| 365/365 [01:24<00:00,  4.33it/s]

RMSE: 0.09198013657135466


In [3]:
# 学習4年分
# model_20221231-4-7.pkl, model_20221231-4-159.pkl
!python ../inference-test/run.py  --exec-path ../inference-test/src --data-dir ../data --start-date 2000 --end-date 2190


start date: 2000, end date:2190

Processing waterlevel data
done
Processing rainfall data
done
Processing tidelevel data
done
Making output data
done

Moved to /notebooks/inference-test/src

Loaded the model.   
100%|█████████████████████████████████████████| 191/191 [00:43<00:00,  4.40it/s]

RMSE: 0.11054407161333896


RMSE: 0.118431486380198  
RMSE: 0.12235856981751622  
RMSE: 0.09385707009578965  
RMSE: 0.11256520782791041  

In [7]:
import os
import shutil

folder_name = "20221231-4"
os.mkdir(os.path.join("../submit-folder/", folder_name))

shutil.copytree('../inference-test/model', os.path.join("../submit-folder", folder_name, "model"))
shutil.copytree('../inference-test/src', os.path.join("../submit-folder", folder_name, "src"))

'../submit-folder/20221231-4/src'

In [8]:
# zip化（事前にrequirements.txtは作成の必要あり）
shutil.make_archive(os.path.join("../submit-folder/Zip", folder_name), 
                    format='zip', 
                    root_dir=os.path.join("../submit-folder", folder_name)
                    )

'/notebooks/submit-folder/Zip/20221231-4.zip'

In [3]:
# 学習5年分
# model_20221231-5-7.pkl, model_20221231-5-159.pkl
!python ../inference-test/run.py  --exec-path ../inference-test/src --data-dir ../data --start-date 1826 --end-date 2190


start date: 1826, end date:2190

Processing waterlevel data
done
Processing rainfall data
done
Processing tidelevel data
done
Making output data
done

Moved to /notebooks/inference-test/src

Loaded the model.   
100%|█████████████████████████████████████████| 365/365 [00:43<00:00,  8.33it/s]

RMSE: 0.08910422396808018


In [2]:
# 学習5年分
# model_20221231-5-7.pkl, model_20221231-5-159.pkl
!python ../inference-test/run.py  --exec-path ../inference-test/src --data-dir ../data --start-date 2000 --end-date 2190


start date: 2000, end date:2190

Processing waterlevel data
done
Processing rainfall data
done
Processing tidelevel data
done
Making output data
done

Moved to /notebooks/inference-test/src

Loaded the model.   
100%|█████████████████████████████████████████| 191/191 [00:25<00:00,  7.62it/s]

RMSE: 0.1083549400695543


RMSE: 0.0907416864775914  
RMSE: 0.10951774485789874

In [4]:
import os
import shutil

folder_name = "20221231-5"
os.mkdir(os.path.join("../submit-folder/", folder_name))

shutil.copytree('../inference-test/model', os.path.join("../submit-folder", folder_name, "model"))
shutil.copytree('../inference-test/src', os.path.join("../submit-folder", folder_name, "src"))

'../submit-folder/20221231-5/src'

In [6]:
# zip化（事前にrequirements.txtは作成の必要あり）
shutil.make_archive(os.path.join("../submit-folder/Zip", folder_name), 
                    format='zip', 
                    root_dir=os.path.join("../submit-folder", folder_name)
                    )

'/notebooks/submit-folder/Zip/20221231-5.zip'

アンサンブル

In [8]:
# 学習4,5年分
!python ../inference-test/run.py  --exec-path ../inference-test/src --data-dir ../data --start-date 1826 --end-date 2190


start date: 1826, end date:2190

Processing waterlevel data
done
Processing rainfall data
done
Processing tidelevel data
done
Making output data
done

Moved to /notebooks/inference-test/src

Loaded the model.   
100%|█████████████████████████████████████████| 365/365 [00:59<00:00,  6.12it/s]

RMSE: 0.08867815826034137


In [7]:
# 学習4,5年分
!python ../inference-test/run.py  --exec-path ../inference-test/src --data-dir ../data --start-date 2000 --end-date 2190


start date: 2000, end date:2190

Processing waterlevel data
done
Processing rainfall data
done
Processing tidelevel data
done
Making output data
done

Moved to /notebooks/inference-test/src

Loaded the model.   
100%|█████████████████████████████████████████| 191/191 [00:31<00:00,  6.02it/s]

RMSE: 0.10750670841747019


RMSE: 0.09062400413978547  
RMSE: 0.10931318063291166  

In [9]:
import os
import shutil

folder_name = "20221231-6"
os.mkdir(os.path.join("../submit-folder/", folder_name))

shutil.copytree('../inference-test/model', os.path.join("../submit-folder", folder_name, "model"))
shutil.copytree('../inference-test/src', os.path.join("../submit-folder", folder_name, "src"))

'../submit-folder/20221231-6/src'

In [10]:
# zip化（事前にrequirements.txtは作成の必要あり）
shutil.make_archive(os.path.join("../submit-folder/Zip", folder_name), 
                    format='zip', 
                    root_dir=os.path.join("../submit-folder", folder_name)
                    )

'/notebooks/submit-folder/Zip/20221231-6.zip'

K-fold

In [13]:
!python ../inference-test/run.py  --exec-path ../inference-test/src --data-dir ../data --start-date 1826 --end-date 2190


start date: 1826, end date:2190

Processing waterlevel data
done
Processing rainfall data
done
Processing tidelevel data
done
Making output data
done

Moved to /notebooks/inference-test/src

Loaded the model.   
100%|█████████████████████████████████████████| 365/365 [01:33<00:00,  3.89it/s]

RMSE: 0.0867389494985939


In [12]:
!python ../inference-test/run.py  --exec-path ../inference-test/src --data-dir ../data --start-date 2000 --end-date 2190


start date: 2000, end date:2190

Processing waterlevel data
done
Processing rainfall data
done
Processing tidelevel data
done
Making output data
done

Moved to /notebooks/inference-test/src

Loaded the model.   
100%|█████████████████████████████████████████| 191/191 [00:49<00:00,  3.83it/s]

RMSE: 0.10545875213138957


RMSE: 0.09062400413978547  
RMSE: 0.10931318063291166  

In [14]:
import os
import shutil

folder_name = "20221231-7"
os.mkdir(os.path.join("../submit-folder/", folder_name))

shutil.copytree('../inference-test/model', os.path.join("../submit-folder", folder_name, "model"))
shutil.copytree('../inference-test/src', os.path.join("../submit-folder", folder_name, "src"))

'../submit-folder/20221231-7/src'

In [15]:
# zip化（事前にrequirements.txtは作成の必要あり）
shutil.make_archive(os.path.join("../submit-folder/Zip", folder_name), 
                    format='zip', 
                    root_dir=os.path.join("../submit-folder", folder_name)
                    )

'/notebooks/submit-folder/Zip/20221231-7.zip'

K-fold+5年分

In [17]:
!python ../inference-test/run.py  --exec-path ../inference-test/src --data-dir ../data --start-date 1826 --end-date 2190


start date: 1826, end date:2190

Processing waterlevel data
done
Processing rainfall data
done
Processing tidelevel data
done
Making output data
done

Moved to /notebooks/inference-test/src

Loaded the model.   
100%|█████████████████████████████████████████| 365/365 [01:50<00:00,  3.30it/s]

RMSE: 0.08672335159133782


In [16]:
!python ../inference-test/run.py  --exec-path ../inference-test/src --data-dir ../data --start-date 2000 --end-date 2190


start date: 2000, end date:2190

Processing waterlevel data
done
Processing rainfall data
done
Processing tidelevel data
done
Making output data
done

Moved to /notebooks/inference-test/src

Loaded the model.   
100%|█████████████████████████████████████████| 191/191 [00:58<00:00,  3.26it/s]

RMSE: 0.10549256859084247


RMSE: 0.0867389494985939  
RMSE: 0.10545875213138957  

In [19]:
import os
import shutil

folder_name = "20221231-8"
os.mkdir(os.path.join("../submit-folder/", folder_name))

shutil.copytree('../inference-test/model', os.path.join("../submit-folder", folder_name, "model"))
shutil.copytree('../inference-test/src', os.path.join("../submit-folder", folder_name, "src"))

'../submit-folder/20221231-8/src'

In [20]:
# zip化（事前にrequirements.txtは作成の必要あり）
shutil.make_archive(os.path.join("../submit-folder/Zip", folder_name), 
                    format='zip', 
                    root_dir=os.path.join("../submit-folder", folder_name)
                    )

'/notebooks/submit-folder/Zip/20221231-8.zip'

### 雨量追加+異常値修正

In [4]:
# 学習4年分
# model_20221231-4-7.pkl, model_20230101-1-159.pkl
!python ../inference-test/run.py  --exec-path ../inference-test/src --data-dir ../data --start-date 1461 --end-date 2190


start date: 1461, end date:2190

Processing waterlevel data
done
Processing rainfall data
done
Processing tidelevel data
done
Making output data
done

Moved to /notebooks/inference-test/src

Loaded the model.   
100%|█████████████████████████████████████████| 730/730 [03:10<00:00,  3.83it/s]

RMSE: 0.11494098406874739


In [3]:
# 学習4年分
# model_20221231-4-7.pkl, model_20221231-4-159.pkl
!python ../inference-test/run.py  --exec-path ../inference-test/src --data-dir ../data --start-date 1642 --end-date 2190


start date: 1642, end date:2190

Processing waterlevel data
done
Processing rainfall data
done
Processing tidelevel data
done
Making output data
done

Moved to /notebooks/inference-test/src

Loaded the model.   
100%|█████████████████████████████████████████| 549/549 [02:23<00:00,  3.81it/s]

RMSE: 0.1188743886827534


In [2]:
# 学習4年分
# model_20221231-4-7.pkl, model_20221231-4-159.pkl
!python ../inference-test/run.py  --exec-path ../inference-test/src --data-dir ../data --start-date 1826 --end-date 2190


start date: 1826, end date:2190

Processing waterlevel data
done
Processing rainfall data
done
Processing tidelevel data
done
Making output data
done

Moved to /notebooks/inference-test/src

Loaded the model.   
100%|█████████████████████████████████████████| 365/365 [01:31<00:00,  4.01it/s]

RMSE: 0.09240037876064136


In [1]:
# 学習4年分
# model_20221231-4-7.pkl, model_20221231-4-159.pkl
!python ../inference-test/run.py  --exec-path ../inference-test/src --data-dir ../data --start-date 2000 --end-date 2190


start date: 2000, end date:2190

Processing waterlevel data
done
Processing rainfall data
done
Processing tidelevel data
done
Making output data
done

Moved to /notebooks/inference-test/src

Loaded the model.   
100%|█████████████████████████████████████████| 191/191 [00:49<00:00,  3.83it/s]

RMSE: 0.11110265952249726


RMSE: 0.11479712375601998 => RMSE: 0.11494098406874739  
RMSE: 0.1185998215875185 => RMSE: 0.1188743886827534  
RMSE: 0.09198013657135466 => RMSE: 0.09240037876064136  
RMSE: 0.11054407161333896 => RMSE: 0.11110265952249726

In [5]:
import os
import shutil

folder_name = "20230101-1"
os.mkdir(os.path.join("../submit-folder/", folder_name))

shutil.copytree('../inference-test/model', os.path.join("../submit-folder", folder_name, "model"))
shutil.copytree('../inference-test/src', os.path.join("../submit-folder", folder_name, "src"))

'../submit-folder/20230101-1/src'

In [6]:
# zip化（事前にrequirements.txtは作成の必要あり）
shutil.make_archive(os.path.join("../submit-folder/Zip", folder_name), 
                    format='zip', 
                    root_dir=os.path.join("../submit-folder", folder_name)
                    )

'/notebooks/submit-folder/Zip/20230101-1.zip'

In [8]:
# 学習5年分
# model_20221231-5-7.pkl, model_20230101-2-159.pkl
!python ../inference-test/run.py  --exec-path ../inference-test/src --data-dir ../data --start-date 1826 --end-date 2190


start date: 1826, end date:2190

Processing waterlevel data
done
Processing rainfall data
done
Processing tidelevel data
done
Making output data
done

Moved to /notebooks/inference-test/src

Loaded the model.   
100%|█████████████████████████████████████████| 365/365 [01:25<00:00,  4.26it/s]

RMSE: 0.08926382929186057


In [7]:
# 学習5年分
# model_20221231-5-7.pkl, model_20230101-2-159.pkl
!python ../inference-test/run.py  --exec-path ../inference-test/src --data-dir ../data --start-date 2000 --end-date 2190


start date: 2000, end date:2190

Processing waterlevel data
done
Processing rainfall data
done
Processing tidelevel data
done
Making output data
done

Moved to /notebooks/inference-test/src

Loaded the model.   
100%|█████████████████████████████████████████| 191/191 [00:44<00:00,  4.33it/s]

RMSE: 0.10843164994289181


RMSE: 0.08910422396808018 => RMSE: 0.08926382929186057  
RMSE: 0.1083549400695543 => RMSE: 0.10843164994289181

In [9]:
import os
import shutil

folder_name = "20230101-2"
os.mkdir(os.path.join("../submit-folder/", folder_name))

shutil.copytree('../inference-test/model', os.path.join("../submit-folder", folder_name, "model"))
shutil.copytree('../inference-test/src', os.path.join("../submit-folder", folder_name, "src"))

'../submit-folder/20230101-2/src'

In [ ]:
# zip化（事前にrequirements.txtは作成の必要あり）
shutil.make_archive(os.path.join("../submit-folder/Zip", folder_name), 
                    format='zip', 
                    root_dir=os.path.join("../submit-folder", folder_name)
                    )

K-fold

In [2]:
!python ../inference-test/run.py  --exec-path ../inference-test/src --data-dir ../data --start-date 1826 --end-date 2190


start date: 1826, end date:2190

Processing waterlevel data
done
Processing rainfall data
done
Processing tidelevel data
done
Making output data
done

Moved to /notebooks/inference-test/src

Loaded the model.   
100%|█████████████████████████████████████████| 365/365 [01:42<00:00,  3.56it/s]

RMSE: 0.08678769920050255


In [1]:
!python ../inference-test/run.py  --exec-path ../inference-test/src --data-dir ../data --start-date 2000 --end-date 2190


start date: 2000, end date:2190

Processing waterlevel data
done
Processing rainfall data
done
Processing tidelevel data
done
Making output data
done

Moved to /notebooks/inference-test/src

Loaded the model.   
100%|█████████████████████████████████████████| 191/191 [00:57<00:00,  3.33it/s]

RMSE: 0.105360555931289


RMSE: 0.0867389494985939 => RMSE: 0.08678769920050255  
RMSE: 0.10545875213138957 => RMSE: 0.105360555931289  

In [3]:
import os
import shutil

folder_name = "20230101-3"
os.mkdir(os.path.join("../submit-folder/", folder_name))

shutil.copytree('../inference-test/model', os.path.join("../submit-folder", folder_name, "model"))
shutil.copytree('../inference-test/src', os.path.join("../submit-folder", folder_name, "src"))

'../submit-folder/20230101-3/src'

In [4]:
# zip化（事前にrequirements.txtは作成の必要あり）
shutil.make_archive(os.path.join("../submit-folder/Zip", folder_name), 
                    format='zip', 
                    root_dir=os.path.join("../submit-folder", folder_name)
                    )

'/notebooks/submit-folder/Zip/20230101-3.zip'

### 雨量追加+異常値修正+特徴量修正

In [6]:
# 学習4年分
# model_20230102-1-7.pkl, model_20230102-1-159.pkl
!python ../inference-test/run.py  --exec-path ../inference-test/src --data-dir ../data --start-date 1461 --end-date 2190


start date: 1461, end date:2190

Processing waterlevel data
done
Processing rainfall data
done
Processing tidelevel data
done
Making output data
done

Moved to /notebooks/inference-test/src

Loaded the model.   
100%|█████████████████████████████████████████| 730/730 [01:56<00:00,  6.26it/s]

RMSE: 0.11406863737696844


In [5]:
# 学習4年分
# model_20230102-1-7.pkl, model_20230102-1-159.pkl
!python ../inference-test/run.py  --exec-path ../inference-test/src --data-dir ../data --start-date 1642 --end-date 2190


start date: 1642, end date:2190

Processing waterlevel data
done
Processing rainfall data
done
Processing tidelevel data
done
Making output data
done

Moved to /notebooks/inference-test/src

Loaded the model.   
100%|█████████████████████████████████████████| 549/549 [01:28<00:00,  6.24it/s]

RMSE: 0.11798032749953155


In [4]:
# 学習4年分
# model_20230102-1-7.pkl, model_20230102-1-159.pkl
!python ../inference-test/run.py  --exec-path ../inference-test/src --data-dir ../data --start-date 1826 --end-date 2190


start date: 1826, end date:2190

Processing waterlevel data
done
Processing rainfall data
done
Processing tidelevel data
done
Making output data
done

Moved to /notebooks/inference-test/src

Loaded the model.   
100%|█████████████████████████████████████████| 365/365 [00:59<00:00,  6.18it/s]

RMSE: 0.09155994452911816


In [3]:
# 学習4年分
# model_20230102-1-7.pkl, model_20230102-1-159.pkl
!python ../inference-test/run.py  --exec-path ../inference-test/src --data-dir ../data --start-date 2000 --end-date 2190


start date: 2000, end date:2190

Processing waterlevel data
done
Processing rainfall data
done
Processing tidelevel data
done
Making output data
done

Moved to /notebooks/inference-test/src

Loaded the model.   
100%|█████████████████████████████████████████| 191/191 [00:30<00:00,  6.37it/s]

RMSE: 0.1103292719411507


RMSE: 0.11494098406874739 => RMSE: 0.11406863737696844  
RMSE: 0.1188743886827534 => RMSE: 0.11798032749953155  
RMSE: 0.09240037876064136 => RMSE: 0.09155994452911816  
RMSE: 0.11110265952249726 => RMSE: 0.1103292719411507

In [7]:
import os
import shutil

folder_name = "20230102-1"
os.mkdir(os.path.join("../submit-folder/", folder_name))

shutil.copytree('../inference-test/model', os.path.join("../submit-folder", folder_name, "model"))
shutil.copytree('../inference-test/src', os.path.join("../submit-folder", folder_name, "src"))

'../submit-folder/20230102-1/src'

In [8]:
# zip化（事前にrequirements.txtは作成の必要あり）
shutil.make_archive(os.path.join("../submit-folder/Zip", folder_name), 
                    format='zip', 
                    root_dir=os.path.join("../submit-folder", folder_name)
                    )

'/notebooks/submit-folder/Zip/20230102-1.zip'

In [2]:
# 学習5年分
# model_20221231-5-7.pkl, model_20230101-2-159.pkl
!python ../inference-test/run.py  --exec-path ../inference-test/src --data-dir ../data --start-date 1826 --end-date 2190


start date: 1826, end date:2190

Processing waterlevel data
done
Processing rainfall data
done
Processing tidelevel data
done
Making output data
done

Moved to /notebooks/inference-test/src

Loaded the model.   
100%|█████████████████████████████████████████| 365/365 [01:03<00:00,  5.76it/s]

RMSE: 0.08855459691798707


In [1]:
# 学習5年分
# model_20221231-5-7.pkl, model_20230101-2-159.pkl
!python ../inference-test/run.py  --exec-path ../inference-test/src --data-dir ../data --start-date 2000 --end-date 2190


start date: 2000, end date:2190

Processing waterlevel data
done
Processing rainfall data
done
Processing tidelevel data
done
Making output data
done

Moved to /notebooks/inference-test/src

Loaded the model.   
100%|█████████████████████████████████████████| 191/191 [00:33<00:00,  5.70it/s]

RMSE: 0.10769903146897099


RMSE: 0.08926382929186057 => RMSE: 0.08855459691798707  
RMSE: 0.10843164994289181 => RMSE: 0.10769903146897099

In [3]:
import os
import shutil

folder_name = "20230102-2"
os.mkdir(os.path.join("../submit-folder/", folder_name))

shutil.copytree('../inference-test/model', os.path.join("../submit-folder", folder_name, "model"))
shutil.copytree('../inference-test/src', os.path.join("../submit-folder", folder_name, "src"))

'../submit-folder/20230102-2/src'

In [5]:
# zip化（事前にrequirements.txtは作成の必要あり）
shutil.make_archive(os.path.join("../submit-folder/Zip", folder_name), 
                    format='zip', 
                    root_dir=os.path.join("../submit-folder", folder_name)
                    )

'/notebooks/submit-folder/Zip/20230102-2.zip'

K-fold

In [9]:
!python ../inference-test/run.py  --exec-path ../inference-test/src --data-dir ../data --start-date 1826 --end-date 2190


start date: 1826, end date:2190

Processing waterlevel data
done
Processing rainfall data
done
Processing tidelevel data
done
Making output data
done

Moved to /notebooks/inference-test/src

Loaded the model.   
100%|█████████████████████████████████████████| 365/365 [02:45<00:00,  2.20it/s]

RMSE: 0.08621839016903075


In [8]:
!python ../inference-test/run.py  --exec-path ../inference-test/src --data-dir ../data --start-date 2000 --end-date 2190


start date: 2000, end date:2190

Processing waterlevel data
done
Processing rainfall data
done
Processing tidelevel data
done
Making output data
done

Moved to /notebooks/inference-test/src

Loaded the model.   
100%|█████████████████████████████████████████| 191/191 [01:28<00:00,  2.15it/s]

RMSE: 0.10497401842371405


RMSE: 0.08678769920050255 => RMSE: 0.08621839016903075     
RMSE: 0.105360555931289 => RMSE: 0.10497401842371405  

In [10]:
import os
import shutil

folder_name = "20230102-3"
os.mkdir(os.path.join("../submit-folder/", folder_name))

shutil.copytree('../inference-test/model', os.path.join("../submit-folder", folder_name, "model"))
shutil.copytree('../inference-test/src', os.path.join("../submit-folder", folder_name, "src"))

'../submit-folder/20230102-3/src'

In [11]:
# zip化（事前にrequirements.txtは作成の必要あり）
shutil.make_archive(os.path.join("../submit-folder/Zip", folder_name), 
                    format='zip', 
                    root_dir=os.path.join("../submit-folder", folder_name)
                    )

'/notebooks/submit-folder/Zip/20230102-3.zip'

K-fold(6年分)

In [13]:
!python ../inference-test/run.py  --exec-path ../inference-test/src --data-dir ../data --start-date 1826 --end-date 2190


start date: 1826, end date:2190

Processing waterlevel data
done
Processing rainfall data
done
Processing tidelevel data
done
Making output data
done

Moved to /notebooks/inference-test/src

Loaded the model.   
100%|█████████████████████████████████████████| 365/365 [02:54<00:00,  2.09it/s]

RMSE: 0.06925087013686951


In [12]:
!python ../inference-test/run.py  --exec-path ../inference-test/src --data-dir ../data --start-date 2000 --end-date 2190


start date: 2000, end date:2190

Processing waterlevel data
done
Processing rainfall data
done
Processing tidelevel data
done
Making output data
done

Moved to /notebooks/inference-test/src

Loaded the model.   
100%|█████████████████████████████████████████| 191/191 [01:33<00:00,  2.05it/s]

RMSE: 0.0817916082982326


In [14]:
import os
import shutil

folder_name = "20230102-4"
os.mkdir(os.path.join("../submit-folder/", folder_name))

shutil.copytree('../inference-test/model', os.path.join("../submit-folder", folder_name, "model"))
shutil.copytree('../inference-test/src', os.path.join("../submit-folder", folder_name, "src"))

'../submit-folder/20230102-4/src'

In [15]:
# zip化（事前にrequirements.txtは作成の必要あり）
shutil.make_archive(os.path.join("../submit-folder/Zip", folder_name), 
                    format='zip', 
                    root_dir=os.path.join("../submit-folder", folder_name)
                    )

'/notebooks/submit-folder/Zip/20230102-4.zip'